File: dsa-practice/maths/problem1.js
const isPrime = (num)=>{
    for(let i=2;i<Math.sqrt(num);i++){
        if(num/i===0){
            return false;
        }
    }
    return true;
}
const showNum = (num)=>{

    if(isPrime(num)){
        console.log("Number is Prime")
    }
    else{
        console.log("Number is not Prime")
    }
}
showNum(2)
showNum(17)

File: dsa-practice/maths/problem1.cpp
#include<iostream>
using namespace std;
int sqrt(int n){
    int result=0;
    bool resultfound = false;
    int i=0;
    while(resultfound==false){
        if(i*i==n){
            result =i;
            resultfound=true;
        }
        i++;
    }
    return result;
}
int main(){
    int n;
    cin>>n;
    bool flag=0;
    for(int i=2;i<sqrt(n);i++){
        if(n%i==0){
            cout<<"Not a prime number";
            flag=1;
            break;
        }
    }
    if(flag==0){
        cout<<"Prime number";
    }
    cout<<endl;
    return 0;
}